backend/routes/auth.py

```
@@ -0,0 +1,193 @@
+            "user": {"username": user_data["username"], "authenticated": True},
+            "message": "Login successful",
+        }
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Login failed: {str(e)}")
+
+
+@router.get("/me")
+async def get_current_user_info(user: Dict = Depends(get_current_user)) -> Dict:
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)